src/test/java/org/minions/devfund/bruno/html/HtmlValidatorTest.java

```diff
@@ -0,0 +1,125 @@
+package org.minions.devfund.bruno.html;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.util.Queue;
+import java.util.Stack;
+
+import static org.minions.devfund.bruno.html.HtmlValidator.isValidHtml;
+
+public class HtmlValidatorTest {
+
+    @Test
+    public void testFile1() {
+        try {
+            Stack<HtmlTag> var1 = this.validateTestFile("test1.html");
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace the synchronized class "Stack" by an unsynchronized one such as "Deque". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1149)